src/libs/actions/Task.ts

```diff
@@ -1266,6 +1274,19 @@ function clearTaskErrors(reportID: string | undefined) {
     });
 }
 
+function getFinishOnboardingTaskOnyxData(taskName: keyof OnyxTypes.IntroSelected): OnyxData {
+    const taskReportID = introSelected?.[taskName];
+    if (taskReportID) {
+        const taskReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${taskReportID}`];
+        if (taskReport) {
```

Comment:
We should also check if the taskReport is not archived before marking it as completed.
More details:
https://github.com/Expensify/App/issues/61006#issuecomment-2838046194